Site migration: progress view adjustments

andfinally opened the pull request:
Note: This change is behind a feature flag and is part of an in-progress larger project. For context, see pbkcP4-8-p2.

#### Changes proposed in this Pull Request

* Replaces illustration.
* Wraps domains in subheader in spans to prevent them breaking across lines.
* Sizes and aligns progress item icons.

![image](https://user-images.githubusercontent.com/1647564/70260353-ffed7f80-1787-11ea-8555-be0cb4a350e6.png)

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Check out this pull request and run Calypso locally, or view it on calypso.live with the flag tools/migrate.
* Viewing calypso.localhost:3000/migrate/, you should see a SiteSelector component allowing you to choose a site to migrate a Jetpack site into.
* When you select a site, you should see another SiteSelector component, listing all the Jetpack sites your account has access to.
* Selecting one of those sites should take you to a confirmation screen.
* Accept the confirmation. You should see a progress screen detailing the steps of the migration.
  * The illustration should not have a coloured bubble.
  * The progress list item icons should be aligned.
  * The domains in the subheader should not break when you resize your viewport.
* When the migration is done you should see a message telling you it is complete.

jonathansadowski commented on client/my-sites/migrate/section-migrate.jsx:
```diff
@@ -225,22 +225,32 @@ class SectionMigrate extends Component {
+			{
+				args: { sourceSiteDomain: sourceSiteDomain, targetSiteDomain: targetSiteDomain },
+				components: {
+					span: <span className="migrate__domain" />,
```
Should we use this same wrapping for the checklist below?

jonathansadowski commented on client/my-sites/migrate/section-migrate.jsx:
```diff
@@ -225,22 +225,32 @@ class SectionMigrate extends Component {
+		const { sourceSite, targetSite, translate } = this.props;
 		const sourceSiteDomain = get( sourceSite, 'domain' );
 		const targetSiteDomain = get( targetSite, 'domain' );
+		const subHeaderText = translate(
```
Thanks for considering the translatability of the strings, but I really thing we should wait to actually wrap them in `translate` calls until the strings are finalized. See p2JRYi-4gH-p2

andfinally commented:
Thanks for the review Jon – I've removed the translation calls and added the non-breaking spans to the domains in the progress list.

I've had to wrap the progress list text in a span to get React to respect the single white space between the text and the domain (see [Ben Nadel's post](https://www.bennadel.com/blog/2880-a-quick-look-at-rendering-white-space-using-jsx-in-reactjs.htm) on the behaviour). For some reason React respects it within a fragment in the subheader text, but not in the list item text.

mattsherman commented:
Accepted.
